### ISSUE TITLE
How to see contract logs in integration tests?

### ISSUE THREAD

**Issue by ryancwalsh** [near/workspaces-rs#175]
I got a tip from Phuong [here](https://near-foundation.slack.com/archives/C02LV4RRTEJ/p1660077955782499?thread_ts=1660008396.697329&cid=C02LV4RRTEJ) but still haven't been able to figure it out.

I tried this: https://github.com/ryancwalsh/donation_matcher_contract/commit/fa9a190b8ce5c9c69bc98a6cd4744588f129b616 and also the lines shown there that are now commented. 

But I got `'Unable to install global subscriber: SetLoggerError(())'` and wasn't able to figure out what that meant.

[Guille | NEAR](https://discord.com/channels/490367152054992913/542945453533036544/1006886616645783654) suggested today that I should propose this in an issue in this repo.

It would be awesome if the docs could clarify how to get my terminal to show logs of everything (not just lines logged from within the tests, but logs from within the contract, too, like unit tests are able to show).

Thanks!

**Comment by ryancwalsh** [comment]
Hi @ChaoticTempest!

I still haven't been able to figure this out and would love more hints from you if you have a moment. 

Getting logs working will be really helpful in me getting unblocked with [this](https://near-foundation.slack.com/archives/C02LV4RRTEJ/p1660079572787729?thread_ts=1660074295.467219&cid=C02LV4RRTEJ) too.

Thanks!

**Comment by austinabell** [comment]
So those are all included within the [logs method of the result from the transaction](https://docs.rs/workspaces/latest/workspaces/result/struct.CallExecutionDetails.html#method.logs) using workspaces. Workspaces doesn't, and shouldn't, print the logs to the console

**Comment by ryancwalsh** [comment]
@austinabell Thanks. I'm curious why Workspaces shouldn't be able to do this. 

In unit tests (rather than Workspaces tests) I'm able to see regular logs in the terminal output. It's super helpful.

And now I'm testing aspects that unit tests can't test (such as account balances before and after transfers), and I'm getting stuck, and my inclination whenever debugging something is to log the heck out of it so that I can drill down and see what it happening line by line.

Thanks.

**Comment by willemneal** [comment]
> Thanks. I'm curious why Workspaces shouldn't be able to do this.

Because the core of workspaces shouldn't preform I/O that isn't required to interact with the NEAR node. You must manually print the logs and then in your tests run with `--nocapture` or run a test through vs-code, which adds this for you.

**Comment by austinabell** [comment]
> @austinabell Thanks. I'm curious why Workspaces shouldn't be able to do this.
> 
> In unit tests (rather than Workspaces tests) I'm able to see regular logs in the terminal output. It's super helpful.
> 
> And now I'm testing aspects that unit tests can't test (such as account balances before and after transfers), and I'm getting stuck, and my inclination whenever debugging something is to log the heck out of it so that I can drill down and see what it happening line by line.
> 
> Thanks.

You can print out the logs if you'd like, it just won't do it by default. If someone wanted to run a performant and async test, blocking to write to stdout would limit the efficiency. Also, some might not want or care about the output.

**Comment by ryancwalsh** [comment]
@austinabell Is there a config option to enable it to work how I was describing (when I don't care about performance and just want convenient thorough debugging)? If we don't have such an option yet, do you think it's a possible feature request? Thanks.

**Comment by ChaoticTempest** [comment]
@ryancwalsh I don't think it's going to be a good feature to add since it adds maintenance burdens. This would likely be a feature flag as to not incur performance costs, which are hard to deprecate in the future if we decide to move away from it. I don't think this feature would get used that much, since it's not easily discoverable either. A simple `println!("{}", result.logs());` is more intuitive since it's clear that we want to explicitly log these results. If it was buried under a feature flag and automatically logged, it would be pretty hard to debug where those logs are coming from

**Comment by ryancwalsh** [comment]
@ChaoticTempest Got it. Thanks!
